src/migrate.js

```
@@ -0,0 +1,42 @@
+// @flow
+
+import Command from 'cli-engine-command'
+import Plugins from './plugins'
+import path from 'path'
+import fs from 'fs-extra'
+
+export default class extends Command {
+  static topic = 'migrate'
+  static description = 'migrate V5 plugins'
+  plugins = new Plugins(this)
+
+  async run () {
+    let pathArry = this.config.dirs.data.split('/')
+    pathArry[pathArry.length - 1] = 'heroku'
+    let v5Data = pathArry.join('/')
+    let file = path.join(v5Data, 'plugins', 'plugins.json')
+    let pljson
+    try {
+      pljson = fs.readJsonSync(file)
+    } catch (err) {}
```

rather than bailing like you do below, I would put the return inside the catch here, and maybe spit out the error to `this.debug(err)` (I think that will show it just in debug mode)

---

and sometimes here what I like to do is nothing if `err.code === 'ENOENT'` but if it is anything else then throw the error. Perhaps in this case it is still best to ignore any error though.